src/auth/KeyRing.cc

```diff
@@ -205,12 +205,12 @@ void KeyRing::decode(bufferlist::const_iterator& bl) {
   __u8 struct_v;
   auto start_pos = bl;
   try {
+    decode_plaintext(start_pos);
```

Comment:
probably we can just use the first `[` as the magic number, which is `0x5b`, as a plaintext keyring file always looks like
```ini
[section-name]
    key = ...
```

and avoid using `0x5b` as the `struct_v` of `KeyRing` in future? so as long as the file starts with `0x5b`, we can assume it's plaintext, otherwise it's binary. and, we can be still have read out of bound or stack-buffer-overflow issues if the binary keyring is malformed.